app/services/content_scraper.rb

```diff
@@ -0,0 +1,32 @@
+# Returns the
+
+class ContentScraper
+  def self.call(url)
+    new(url).content
+  end
+
+  attr_reader :url
+
+  def initialize(url)
+    @url = url
+  end
+
+  def content
+    case headers["content-type"]
```

Comment:
what happens here if the content type is something different? Was wondering if we should have a spec for that as well?

Reply:
At the moment it will return nothing. Yes - you are right - there should be a spec for that!

To be honest, I'm not sure at the moment if this is the right strategy. I was thinking of defaulting to HTML and also wondered about raising an error. The main problem with the latter being that if this is run offline then reporting back to the user that there was a problem is difficult.

So this is definitely work in progress.

Reply:
I've added a comment and additional test, which I think clarifies what happens if the content type is not known.

I think this may be the best solution (but in another PR):

- Add a status flag to the `Item`. The statuses could be:
  - **created** initial flag on creation
  - **indexed** after successful index completion
  - **error** Error
- Add an error store. This could be an attribute of Item or a separate object.